frontend/src/components/UserDropdown.tsx

```diff
@@ -71,41 +78,61 @@ const NoResults = styled.div`
   padding: 10px;
 `;
 
-const Overlay = styled.div`
-  width: 100vw;
-  height: 100vh;
-  top: 70px;
-  left: 0;
-  right: 0;
-  bottom: 0;
-  position: fixed;
-  background: rgbd(0, 0, 0, 0);
-  z-index: -1;
-`;
-
 type Option = User | RenterCandidate;
 
 type SelectProps = {
+  width?: string;
   placeholder: string;
+  initialSelection?: Option;
   options: Option[];
-  onSelect?: (value: Option | undefined) => void; //callback function for parent, sends current selected user
+  onSelect: (value: Option) => void; //callback function for parent, sends current selected user
   reset?: boolean;
   isRCDropdown?: boolean;
 };
 
-export function UserDropdown({ placeholder, options, onSelect, reset, isRCDropdown }: SelectProps) {
+export function UserDropdown({
+  width,
+  placeholder,
+  initialSelection,
+  options,
+  onSelect,
+  reset,
+  isRCDropdown,
+}: SelectProps) {
   const [openMenu, setOpenMenu] = useState(false);
   const [searchValue, setSearchValue] = useState(""); //current text value in select input box
   const [validOptions, setValidOptions] = useState<Option[]>(options); //all RS filtered through search
-  const [solid, setSolid] = useState(false); //search text color
   const [currentSelected, setCurrentSelected] = useState<Option>(); //current selected RS
 
   const handleSelect = (selectedValue: Option) => {
-    setSearchValue(selectedValue.firstName + " " + selectedValue.lastName);
     setCurrentSelected(selectedValue);
+    onSelect(selectedValue);
```

Comment:
@Azhou2023 FYI I updated the functionality of `UserDropdown` so that the user must click an option to trigger the `onSelect` callback. Typing their name does not automatically select the user anymore.

Reply:
Sounds good!